app/controllers/auth/AuthJourney.scala

```diff
@@ -17,24 +17,32 @@
 package controllers.auth
 
 import com.google.inject.ImplementedBy
+import controllers.auth.actions._
+import controllers.auth.requests.{AuthenticatedRequest, PayeAuthenticatedRequest}
 import play.api.mvc.{ActionBuilder, AnyContent}
 
 import javax.inject.Inject
 
 @ImplementedBy(classOf[AuthJourneyImpl])
 trait AuthJourney {
-  val authWithSelfAssessment: ActionBuilder[AuthenticatedRequest, AnyContent]
-  val authWithSingleGGCheck: ActionBuilder[PayeAuthenticatedRequest, AnyContent]
+  val authMinimal: ActionBuilder[AuthenticatedRequest, AnyContent]
+  val authForIndividualsAndAgents: ActionBuilder[AuthenticatedRequest, AnyContent]
+  val authForSAIndividualsAndAgentsOnly: ActionBuilder[AuthenticatedRequest, AnyContent]
+  val authForPayeIndividualsOnly: ActionBuilder[PayeAuthenticatedRequest, AnyContent]
 }
 
 class AuthJourneyImpl @Inject() (
+  minAuthAction: MinAuthAction,
   authAction: AuthAction,
-  selfAssessmentAction: SelfAssessmentAction,
-  payeAuthAction: PayeAuthAction,
-  pertaxAuthAction: PertaxAuthAction
+  payeAuthAction: PayeAuthAction
 ) extends AuthJourney {
-  override val authWithSelfAssessment: ActionBuilder[AuthenticatedRequest, AnyContent]    =
-    authAction andThen selfAssessmentAction
-  override val authWithSingleGGCheck: ActionBuilder[PayeAuthenticatedRequest, AnyContent] =
-    payeAuthAction andThen pertaxAuthAction
+  override val authMinimal: ActionBuilder[AuthenticatedRequest, AnyContent] = minAuthAction
+
+  override val authForIndividualsAndAgents: ActionBuilder[AuthenticatedRequest, AnyContent] =
+    authAction(saShutterCheck = false, agentTokenCheck = true, utrCheck = false)
+
+  override val authForSAIndividualsAndAgentsOnly: ActionBuilder[AuthenticatedRequest, AnyContent] =
+    authAction(saShutterCheck = true, agentTokenCheck = false, utrCheck = true)
+
+  override val authForPayeIndividualsOnly: ActionBuilder[PayeAuthenticatedRequest, AnyContent] = payeAuthAction
```

Comment:
Names not clear.
If agent only why individuals?

Reply:
One at line 44 is meant to indicate auth for SA - which could be individuals or agents. 
One at line 47 is meant to indicate auth for PAYE only - which can be individuals only
If you want to suggest alternative names I'm happy to change them if it's not clear

Reply:
authForSAIndividualsOrAgents, authForPayeIndividuals?

Reply:
Done